### ISSUE TITLE
0.1.7 broke AWS CodeArtifact

### ISSUE THREAD

**Issue by bra-fsn** [astral-sh/uv#1860]
I use AWS CodeArtifact to store/get local Python packages.
After getting a token for it and specifying the URL with uv's `--extra-index-url`, I can install those packages with `uv pip install $pkg_name --extra-index-url "$URL"` with uv version `0.1.6`, but I get the following error using `0.1.7`:
```
uv pip install $pkg_name --extra-index-url "$URL"
error: Failed to download: $pkg_name==1.2.5
  Caused by: HTTP status client error (401 Unauthorized) for url (https://XXX-packages-XXX.d.codeartifact.us-west-2.amazonaws.com/pypi/python/simple/$pkg_name/1.2.5/$pkg_name-1.2.5-py3-none-any.whl#sha256=f168c68d34f2404c9d9289406a9e340ede6224888baceba18dfb980912fe73ae)
```

**Comment by zanieb** [comment]
I can confirm this is a regression from 0.1.6 to 0.1.7, thanks for the report!

**Comment by zanieb** [comment]
This should be resolved in v0.1.8
